docs/pages/bare/using-expo-client.md

```diff
@@ -1,49 +1,69 @@
 
-## Practical patterns for client-compatible bare apps
+Now you can launch your app in Expo Go by pressing `i` or `a` in the Terminal UI. Some features may cause your app to throw errors because certain native code is missing, continue reading to learn how you can conditionally skip unsupported APIs.
 
-### Prefer `npx expo install` over `npm install` to add Expo SDK packages
+> Unlike `npx react-native start` the command `npx expo start` hosts an app manifest that dev clients like Expo Go can use to load arbitrary projects. Think of an app manifest like the `<head />` element of an `index.html` but for React Native apps. To view this manifest, visit the dev server URL in your web browser.
 
-This will ensure that you get a version of the package that is compatible with the SDK version in your app. If you use `npm install` directly instead, you may end up with a newer version of the package that isn't supported in Expo Go yet.
+## Installing libraries
 
-### Use conditional inline requires to provide fallbacks
+You'll want to ensure your app is using the most compatible library versions for the project's `react-native` version, this means using `npx expo install` instead of `npm install`. Learn more in [`npx expo install`](/workflow/expo-cli#install).
 
-Picture this scenario: you need a beautiful map in your app and Google Maps just won't cut it, so you add '@react-native-mapbox-gl/maps'. Expo doesn't include this in the SDK, so you can't run any code that imports it in the Expo Go app. You can handle this by wrapping `MapView` access with a wrapper that provides a fallback in Expo Go, and otherwise uses the native Mapbox library:
+## Runtime detection
+
+The easiest way to detect where the JavaScript bundle is running is to check the [`Constants.executionEnvironment`](/versions/latest/sdk/constants/#nativeconstants--properties).
+
+<Terminal cmd={[
+"$ npx expo install expo-constants",
+]} cmdCopy="npx expo install expo-constants" />
+
+```tsx
+import Constants, { ExecutionEnvironment } from 'expo-constants';
+
+// `true` when running in Expo Go.
+const isExpoGo = Constants.executionEnvironment === ExecutionEnvironment.StoreClient;
+```
+
+You can use this boolean to conditionally require custom native code. Here's an example using the library `react-native-blurhash` which is not available in the Expo Go app:
+
+<SnackInline dependencies={['expo-constants', 'react-native-blurhash']}>
 
 ```js
-// MapView.js
-import * as React from 'react';
+import React from 'react';
 import { StyleSheet, Text, View } from 'react-native';
-import Constants from 'expo-constants';
+import Constants, { ExecutionEnvironment } from 'expo-constants';
+
+// `true` when running in Expo Go.
+const isExpoGo = Constants.executionEnvironment === ExecutionEnvironment.StoreClient;
 
-let MapView;
+let Blurhash;
 
```

Comment:
```suggestion
// Create a fallback for Expo Go
```

Can we add a comment here about creating a fallback for Expo Go? I think it will be a nice addition.